tests/Casts/DateTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Tests\Casts;
+
+use Illuminate\Support\Carbon;
+use MongoDB\Laravel\Tests\Models\Casting;
+use MongoDB\Laravel\Tests\TestCase;
+
+use function now;
+
+class DateTest extends TestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        Casting::truncate();
+    }
+
+    public function testDate(): void
+    {
+        $model = Casting::query()->create(['dateField' => now()]);
```

Comment:
Would it make sense to add a test with `\DateTime`? It actually works.

Reply:
Done in c715194.